### ISSUE TITLE
Implementation of the StorageProvider interface

### ISSUE THREAD

**Issue by Wikinaut** [nextcloud/android#2258]
According to the answer to my issue https://github.com/bitcoin-wallet/bitcoin-wallet/issues/519 it seems that Nextcloud Android is not yet compliant to https://github.com/googlesamples/android-StorageProvider .

Please can someone have a look into this issue, thanks.

**Comment by tobiasKaminsky** [comment]
We already have a DocumentProvider.
It needs to be clarified if StorageProvider can replace DocumentProvider, how it behaves on <4.4 devices.

**Comment by Wikinaut** [comment]
@tobiasKaminsky Thanks for swift reply. Perhaps you contact the https://github.com/bitcoin-wallet/bitcoin-wallet/issues/519 developer to check what needs to be changed and where. Otherwise, currently, it is not possible to select Nextcloud as destination for a wallet-security backup - which was possible earlier.

**Comment by tobiasKaminsky** [comment]
With upcoming version (and current dev version) this should be fixed/included, please test it and if there is something not working as expected, please re-open.

**Comment by MisterY** [comment]
I have v3.5.0 installed but don't see NextCloud in the list of storage providers. I wonder what is missing. The app was installed from F-Droid. Other users report that they do see it in the list. 
Normally I have device credentials protection on NextCloud app. Tried removing it but still did not show. 
Just reporting this FYI.

**Comment by tobiasKaminsky** [comment]
Sometimes NC is not shown in list. Then it helped for me to kill the other app (e.g. documents) to force "scan"/recreate the storage provider list.
Also a reboot helped…
All in all it is more a problem of the "other" app not recognizing NC, instead of NC not registering itself as storage provider.
(if you have enabled protection, it will not be shown)

**Comment by 1989gironimo** [comment]
> (if you have enabled protection, it will not be shown)

Why is this the case?
 @tobiasKaminsky

**Comment by rhclayto** [comment]
I found that if the Nexcloud Android app has a passcode set, your Nextcloud storage account will not show up as a storage location anywhere in Android. I discovered this while trying to set up the Seedvault backup app. In the settings of Seedvault it showed me that Nextcloud was a potential storage location, but that the passcode must be removed in order for it to work. When I removed the passcode, Nextcloud thereupon also began showing up in storage location selection menus (file save location, etc.). Only the Seedvault app actually told me what was causing the problem. I hope this helps somebody else.
